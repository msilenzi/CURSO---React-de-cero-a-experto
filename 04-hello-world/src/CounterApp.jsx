import PropTypes from 'prop-types'
import { useState } from 'react'

function CounterApp({ initialValue }) {
  const [counter, setCounter] = useState(initialValue)

  function increaseCounter() {
    setCounter(counter + 1)
  }

  function decreaseCounter() {
    setCounter(counter - 1)
  }

  function resetCounter() {
    setCounter(initialValue)
  }

  return (
    <>
      <h1>Counter</h1>
      <h2 className="counter">{counter}</h2>
      <div className="counter__buttons">
        <button onClick={decreaseCounter}>-</button>
        <button onClick={resetCounter}>reset</button>
        <button onClick={increaseCounter}>+</button>
      </div>
    </>
  )
}

CounterApp.propTypes = {
  initialValue: PropTypes.number.isRequired,
}

export default CounterApp
