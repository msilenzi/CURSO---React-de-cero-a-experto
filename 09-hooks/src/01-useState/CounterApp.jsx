import PropTypes from 'prop-types'
import { useState } from 'react'

function CounterApp() {
  const [{ counter1, counter2, counter3 }, setCounter] = useState({
    counter1: 0,
    counter2: 0,
    counter3: 0,
  })

  return (
    <div style={{ width: '320px' }}>
      <Counter
        name="Counter 1"
        value={counter1}
        update={() =>
          setCounter((prev) => ({ ...prev, counter1: counter1 + 1 }))
        }
      />
      <hr className="my-4" />
      <Counter
        name="Counter 2"
        value={counter2}
        update={() =>
          setCounter((prev) => ({ ...prev, counter2: counter2 + 1 }))
        }
      />
      <hr className="my-4" />{' '}
      <Counter
        name="Counter 3"
        value={counter3}
        update={() =>
          setCounter((prev) => ({ ...prev, counter3: counter3 + 1 }))
        }
      />
    </div>
  )
}

function Counter({ name, value, update }) {
  return (
    <div className="d-flex justify-content-between align-items-center">
      <h3 className="m-0">
        {name}: <b>{value}</b>
      </h3>
      <button className="btn btn-dark" onClick={update}>
        +1
      </button>
    </div>
  )
}

Counter.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.number.isRequired,
  update: PropTypes.func.isRequired,
}

export default CounterApp
