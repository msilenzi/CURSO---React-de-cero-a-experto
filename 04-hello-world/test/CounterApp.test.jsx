import { render, screen, fireEvent } from '@testing-library/react'
import CounterApp from '../src/CounterApp'

describe('Pruebas en <CounterApp />', () => {
  test('debe coincidir con el snapshot', () => {
    const { container } = render(<CounterApp initialValue={100} />)
    expect(container).toMatchSnapshot()
  })

  test('debe mostrar el valor inicial de 100', () => {
    render(<CounterApp initialValue={100} />)
    expect(screen.getByRole('heading', { level: 2 }).innerHTML).toBe('100')
  })

  test('debe incrementar el contador', () => {
    render(<CounterApp initialValue={10} />)
    fireEvent.click(screen.getByRole('button', { name: '+' }))
    expect(screen.getByRole('heading', { level: 2 }).innerHTML).toBe('11')
  })

  test('debe decrementar el contador', () => {
    render(<CounterApp initialValue={10} />)
    fireEvent.click(screen.getByRole('button', { name: '-' }))
    expect(screen.getByRole('heading', { level: 2 }).innerHTML).toBe('9')
  })

  test('debe resetear el contador', () => {
    render(<CounterApp initialValue={10} />)
    
    fireEvent.click(screen.getByRole('button', { name: '+' }))
    fireEvent.click(screen.getByRole('button', { name: '+' }))
    fireEvent.click(screen.getByRole('button', { name: '+' }))

    fireEvent.click(screen.getByRole('button', { name: 'reset' }))

    expect(screen.getByRole('heading', { level: 2 }).innerHTML).toBe('10')
  })
})
