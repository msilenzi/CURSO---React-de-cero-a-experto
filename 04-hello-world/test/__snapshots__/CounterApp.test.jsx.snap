// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> debe coincidir con el snapshot 1`] = `
<div>
  <h1>
    Counter
  </h1>
  <h2
    class="counter"
  >
    100
  </h2>
  <div
    class="counter__buttons"
  >
    <button>
      -
    </button>
    <button>
      reset
    </button>
    <button>
      +
    </button>
  </div>
</div>
`;
